pkg/egresspolicy/manager.go

```diff
@@ -0,0 +1,461 @@
+// returns bool indicates if policy is added, err inidates first encountered error
+func (manager *Manager) AddEgressPolicy(config EgressPolicyConfig) (bool, error) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+
+	_, ok := manager.policyConfigs[config.id]
+	if ok {
+		log.WithField(
+			logfields.CiliumEgressNATPolicyName, config.id.Name).Warn("CiliumEgressNATPolicy already exists and is not re-added")
+		return false, fmt.Errorf("%v already exists", config.id.Name)
+	}
+
+	err := manager.isValidConfig(config)
+	if err != nil {
+		return false, fmt.Errorf("invalid config: %w", err)
+	}
+
+	manager.policyConfigs[config.id] = &config
+	pods := manager.getLocalPodsForPolicy(&config)
+	err = manager.upsertPolicyPod(&config, pods...)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// Returns a slice of endpoint pods metadata that are selected by the given policy config.
+func (manager *Manager) getLocalPodsForPolicy(config *EgressPolicyConfig) []*podMetadata {
+	var (
+		retPods []*podMetadata
+		podData *podMetadata
+		err     error
+	)
+
+	podStore := manager.storeGetter.GetStore("pod")
+	for _, podItem := range podStore.List() {
+		pod, ok := podItem.(*slim_corev1.Pod)
+		if !ok {
+			continue
+		}
+		podIPs := k8sUtils.ValidIPs(pod.Status)
+		if len(podIPs) == 0 {
+			continue
+		}
+		if podData, err = manager.getPodMetadata(pod, podIPs); err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.K8sPodName:   pod.Name,
+				logfields.K8sNamespace: pod.Namespace,
+			}).Error("Failed to get valid pod metadata")
+			continue
+		}
+		if k8sUtils.GetLatestPodReadiness(pod.Status) != slim_corev1.ConditionTrue {
+			continue
+		}
+		if !config.policyConfigSelectsPod(podData) {
+			continue
+		}
+		retPods = append(retPods, podData)
+	}
+
+	return retPods
+}
+
+// DeleteEgressPolicy is the event handler for policy deletions.
+// It deletes the internal state associated with the given policy, including egress eBPF map entries.
+func (manager *Manager) DeleteEgressPolicy(config EgressPolicyConfig) error {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+
+	storedConfig := manager.policyConfigs[config.id]
+	if storedConfig == nil {
+		return fmt.Errorf("CiliumEgressNATPolicy delete error: policy not found")
+	}
+	log.WithFields(logrus.Fields{"policyID": config.id}).
+		Debug("Delete local egress policy")
+
+	for podId, policies := range manager.policyPods {
+		var newPolicyList []policyID
+		// make a new list excluding policy that is to be deleted
+		for _, policyId := range policies {
+			if policyId == storedConfig.id {
+				// found policy to pod mapping, need to delete egress map entry
+				// identified by pod and config
+				if podData, ok := manager.podDataStore[podId]; ok {
+					err := manager.deleteEgressMap(storedConfig, podData)
+					if err != nil {
+						return err
+					}
+				} else {
+					return fmt.Errorf("Failed to get pod data for %v", podId)
+				}
```

Comment:
We can simplify the if-else branching here

```suggestion
                              podData, ok := manager.podDataStore[podId]
				if !ok {
				    return fmt.Errorf("failed to get pod data for %v", podId)
				}
				if err := manager.deleteEgressMap(storedConfig, podData); err != nil {
				    return err
				}
```

Reply:
Fixed.